pkg/scraper/cache.go

```diff
@@ -0,0 +1,296 @@
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			if len(via) >= maxRedirects {
+				return fmt.Errorf("%w: gave up after %d redirects", ErrMaxRedirects, maxRedirects)
+			}
+			return nil
+		},
+	}
+
+	return client
+}
+
+// NewCache returns a new Cache loading scraper configurations from the
+// scraper path provided in the global config object. It returns a new
+// instance and an error if the scraper directory could not be loaded.
+//
+// Scraper configurations are loaded from yml files in the provided scrapers
+// directory and any subdirectories.
+func NewCache(globalConfig GlobalConfig, txnManager models.TransactionManager) (*Cache, error) {
+	// HTTP Client setup
+	client := newClient(globalConfig)
+
+	scrapers, err := loadScrapers(globalConfig, txnManager)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Cache{
+		client:       client,
+		globalConfig: globalConfig,
+		scrapers:     scrapers,
+		txnManager:   txnManager,
+	}, nil
+}
+
+func loadScrapers(globalConfig GlobalConfig, txnManager models.TransactionManager) (map[string]scraper, error) {
+	path := globalConfig.GetScrapersPath()
+	scrapers := make(map[string]scraper)
+
+	// Add built-in scrapers
+	freeOnes := getFreeonesScraper(txnManager, globalConfig)
+	autoTag := getAutoTagScraper(txnManager, globalConfig)
+	scrapers[freeOnes.spec().ID] = freeOnes
+	scrapers[autoTag.spec().ID] = autoTag
+
+	logger.Debugf("Reading scraper configs from %s", path)
+
+	scraperFiles := []string{}
+	err := utils.SymWalk(path, func(fp string, f os.FileInfo, err error) error {
+		if filepath.Ext(fp) == ".yml" {
+			c, err := loadConfigFromYAMLFile(fp)
+			if err != nil {
+				logger.Errorf("Error loading scraper %s: %v", fp, err)
+			} else {
+				scraper := newGroupScraper(*c, txnManager, globalConfig)
+				scrapers[scraper.spec().ID] = scraper
+			}
+			scraperFiles = append(scraperFiles, fp)
+		}
+		return nil
+	})
+
+	if err != nil {
+		logger.Errorf("Error reading scraper configs: %v", err)
+		return nil, err
+	}
+
+	return scrapers, nil
+}
+
+// ReloadScrapers clears the scraper cache and reloads from the scraper path.
+// In the event of an error during loading, the cache will be left empty.
+func (c *Cache) ReloadScrapers() error {
+	c.scrapers = nil
+	scrapers, err := loadScrapers(c.globalConfig, c.txnManager)
+	if err != nil {
+		return err
+	}
+
+	c.scrapers = scrapers
+	return nil
+}
+
+// TODO - don't think this is needed
+// UpdateConfig updates the global config for the cache. If the scraper path
+// has changed, ReloadScrapers will need to be called separately.
+func (c *Cache) UpdateConfig(globalConfig GlobalConfig) {
+	c.globalConfig = globalConfig
+}
+
+// ListScrapers returns scrapers matching a given kind
+func (c Cache) ListScrapers(k models.ScrapeContentType) []*models.Scraper {
+	var ret []*models.Scraper
+	for _, s := range c.scrapers {
+		if s.supports(k) {
+			spec := s.spec()
+			ret = append(ret, &spec)
+		}
+	}
+
```

Comment:
It sees that the `order` changes from the latest commit need to be replicated here for the `deprecated` listSceneScrapers/listPerformerScrapers/... functions to return a sorted list

Reply:
Yep, because the underlying cache now uses a map. I'll get those sorted as well.